src/main/core/scheduler/scheduler.c

```diff
@@ -609,8 +615,14 @@ void scheduler_start(Scheduler* scheduler) {
     g_mutex_unlock(&scheduler->globalLock);
 
     if(scheduler->policyType != SP_SERIAL_GLOBAL) {
-        /* this will cause a worker to execute the locked initialization in awaitStart */
+        /* this will cause all workers to start their hosts in awaitStart */
         countdownlatch_countDownAwait(scheduler->startBarrier);
+        /* we wait for the workers to finish starting hosts before preparing
+         * round 1 */
+        countdownlatch_countDownAwait(scheduler->finishBarrier);
+        /* reset the barriers so we can use them later */
+        countdownlatch_reset(scheduler->startBarrier);
```

Comment:
It doesn't look like `startBarrier` is reused, so this shouldn't be needed? (And same for `finishBarrier` if you end up not reusing it)